### PR TITLE
Fix the light client protocol protobuf schema

### LINKED ISSUE #12727: Substrate sometimes sends back an empty response to light client requests
### Is there an existing issue?

- [ ] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

cc https://github.com/paritytech/smoldot/issues/2975

I don't know under which condition (I don't know if it's a new version or if the node is overwhelmed or something), but Substrate sometimes sends back an empty response to light client requests. By "empty response" I mean that it sends back a `0` byte indicating that the response is empty, then closes the substream as if the response had finished being sent back.

It's really not supposed to do that. In case where the node is overwhelmed, Substrate is supposed to close the substream without sending any response at all without even any length (as otherwise sending back an empty response would be problem in case the response is legitimately empty).

### Steps to reproduce

I don't know, unfortunately

### PR DESCRIPTION
**Pull request by tomaka**
Fix #12727 
Amends https://github.com/paritytech/substrate/pull/12372

This PR fixes https://github.com/paritytech/substrate/pull/12372 to do what it was intended to do. Right now the light client handler sends back an empty response in case of error, while what I intended was that it sends a response (most importantly, indicating the type of response) but without a proof in it. This is what this PR does.

This PR also switches the schema to protobuf 2.
In protobuf 2, each field is either optional or required. This is good. It's explicit. You know exactly what you get on the wire. Google, the best™ company in the world, then saw that sometimes we send one or two extra bytes that aren't strictly necessary, and, because they have too many engineers and too much time, decided that they were going to release protobuf 3 where they know better than you. They made fields optional by default, and magically omitted fields from messages when the fields were present but equal to their default value. In other words, a lot of magic.
This confused a lot of people, and Google seemingly realized that sometimes it wasn't a good idea to do so, so they are now adding back an `optional` keyword to protobuf 3, and because it doesn't mean the same thing as "optional" in protobuf 2, it probably confuses people even more. This keyword could have been used in this PR, unfortunately it was unstable until recently and requires passing an option to older protobuf compilers, which we really don't want to bother with here.
You can read more at https://github.com/protocolbuffers/protobuf/blob/main/docs/field_presence.md if you are into self-harm.

So, long story short, I've switched back to protobuf 2. Libp2p as a whole still uses protobuf 2 as an example. The only reason why we've switched to protobuf 3 as far as I recall is "bigger number is better", and not anything in particular.

### CODE REVIEW

**Comment by tomaka** [comment]
I took the liberty to remove the `Pair` struct from the protobuf schema, as it is unused.

**Review by dmitry-markin — APPROVED** [review]

**Review by altonen — APPROVED** [review]

**Comment by tomaka** [comment]
bot merge

**Comment by jasl** [comment]
ProtoC 3.15 re-add the `optional` keyward, so basically the behavior is just like proto2, but it's too new that most distribution not bundle that
